### PR TITLE
adds reusable FormInputs and buttons

### PR DESCRIPTION
**Pull request by Joggz**
# What does this PR do?
- adds reusable inputs and buttons.

#ScreenShots
- 
reusauable forms and buttons

### CODE REVIEW

**Review by John-pels — COMMENTED** [review]
The buttons are still not accurate both in colors and props. You are not to hardcode the props yourself, just pass in the appropriate props, add the necessary styles and leave that to whoever that wants it to pick from the styles,size,text and colors.

**Review by Mlg29 — COMMENTED** [review]
-The text color is kind of not okay for the sign in with facebook
-Also, I don't think the buttons are responsive

**Comment by Joggz** [comment]
> The buttons are still not accurate both in colors and props. You are not to hardcode the props yourself, just pass in the appropriate props, add the necessary styles and leave that to whoever that wants it to pick from the styles,size,text and colors.

Thanks for making out time to review, I actually did not intend to make it seem like I hardcoded the button styles, I was only intuitively trying to show how the props value can be passed in and intended to show that both components are reusable. I will take out the Button components in the app.js file.

**Comment by Joggz** [comment]
> -The text color is kind of not okay for the sign in with facebook
> -Also, I don't think the buttons are responsive

Thanks for thanking out time to review, Yh you are right, but about the color, I only intend to show the button component is reusable, about the responsiveness I think it will have to do with its parent element

File: dae-alright/src/components/Button/Button.component.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <button className={`${className}`} style={{
+        backgroundColor:bgColor,
+        width: width,
+        height: height,
+        color: color,
+        padding: '3px'}}>
+
+        { icons } 
+        
+        { btnDetails }
```
Comment (SEUNAGBEYE) [inline dae-alright/src/components/Button/Button.component.jsx]: `btnDetails` isn't a good variable name, I think you should call this `text`

File: dae-alright/src/components/Button/Button.component.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import './Button.style.scss';
+
+const Button = ({btnDetails,bgColor, color, icons, width, height, className}) => (
```
Comment (SEUNAGBEYE) [inline dae-alright/src/components/Button/Button.component.jsx]: I think four props are needed here, `text`, `style`, and `className`

File: dae-alright/src/components/Button/Button.component.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    <button className={`${className}`} style={{
+        backgroundColor:bgColor,
```
Comment (SEUNAGBEYE) [inline dae-alright/src/components/Button/Button.component.jsx]: You can do `style={{...style, padding: '3px'}}`

File: dae-alright/src/components/Button/Button.component.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        width: width,
+        height: height,
+        color: color,
+        padding: '3px'}}>
+
+        { icons } 
+        
+        { btnDetails }
+     </button>
+    //   <Button btnDetails={' Sign in with facebook'} className={'button-lg'} color={'steelblue'} icons={<FontAwesomeIcon icon={faFacebookSquare}  size='lg'/>} bgColor={'red'}/>
```
Comment (SEUNAGBEYE) [inline dae-alright/src/components/Button/Button.component.jsx]: Remove the dead code here

File: dae-alright/src/components/Button/Button.style.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+.button-sm{
+    display: block;
+    margin: 5px;
+    width: 7rem;
+    height: 2rem;
+    background-color: hsl(62,67,56);
+    border: none;
+    outline: none;
+    color: white;
+}
```
Comment (SEUNAGBEYE) [inline dae-alright/src/components/Button/Button.style.scss]: add a new line here

File: dae-alright/src/components/FormInput/FormInput.component.jsx [hunk 6]
```diff
@@ -1,9 +1,17 @@
...
+const FormInput = ({placeholder, type, height, width}) => {
+    return(
+    <input className='input px-4'
+    type={type} 
+    placeholder={placeholder} style={{
+      height: height,
+      width: width,
+
+    }}/>
+    // <FormInput type={'password'} placeholder={'password'} height={'48px'} width={'15rem'}/>
```
Comment (SEUNAGBEYE) [inline dae-alright/src/components/FormInput/FormInput.component.jsx]: Remove this

File: dae-alright/src/components/FormInput/FormInput.component.jsx [hunk 7]
```diff
@@ -1,9 +1,17 @@
...
-  </div>
-)
+const FormInput = ({placeholder, type, height, width}) => {
+    return(
+    <input className='input px-4'
+    type={type} 
+    placeholder={placeholder} style={{
+      height: height,
+      width: width,
+
```
Comment (SEUNAGBEYE) [inline dae-alright/src/components/FormInput/FormInput.component.jsx]: This needs an onChange event handler

File: dae-alright/src/App.js [hunk 8]
```diff
@@ -1,14 +1,27 @@
...
       <h1>dae-Alright</h1>
+      <Button btnDetails={' log in'} className={'button-lg'} />
+      <Button btnDetails={' Sign in with facebook'} className={'button-lg'} color={'steelblue'} icons={<FontAwesomeIcon icon={faFacebookSquare}  size='lg'/>} bgColor={'red'}/>
+      <Button btnDetails={'Sign in with facebook'} className={'button-lg'} color={'tomato'} icons={<FontAwesomeIcon icon={faGooglePlusG}  size='lg'/>}  bgColor={'purple'}/>
+      <Button btnDetails={' Proceed to payment'} className={'button-xlg'}   />
+      <Button btnDetails={' Find Food'} className={'button-sm'}   />
+      <Button btnDetails={' Find Food'} className={'button-sm'}  bgColor={'whitesmoke'} color={'darkkhaki'} />
+      <FormInput type={'email'} placeholder={'Email'}/>
+      <FormInput type={'password'} placeholder={'password'} height={'48px'} width={'15rem'}/>
+      <FormInput type={'email'} placeholder={'Your Input an be anything'} />
```
Comment (SEUNAGBEYE) [inline dae-alright/src/App.js]: Remove these components here when you're done testing.

**Review by SEUNAGBEYE — CHANGES_REQUESTED** [review]
Add `prop types validation` to your component and let's get this merged ASAP because it has a higher priority.

**Comment by SEUNAGBEYE** [comment]
Also, rename your `Button.component.jsx` and `FormInput.component.jsx` to both `index.jsx` for easy import like

```
import Button from src/components/Button
```

File: dae-alright/src/components/Button/Button.component.jsx (hunk 1 above)
Comment (Joggz) [reply dae-alright/src/components/Button/Button.component.jsx]: Done. commit link : aae0f32ce7a3bebbdea9aa241f946689d2bf21c8

File: dae-alright/src/components/Button/Button.component.jsx (hunk 2 above)
Comment (Joggz) [reply dae-alright/src/components/Button/Button.component.jsx]: Done.

File: dae-alright/src/components/Button/Button.component.jsx (hunk 3 above)
Comment (Joggz) [reply dae-alright/src/components/Button/Button.component.jsx]: Done, Commit link: aae0f32ce7a3bebbdea9aa241f946689d2bf21c8

File: dae-alright/src/components/Button/Button.component.jsx (hunk 4 above)
Comment (Joggz) [reply dae-alright/src/components/Button/Button.component.jsx]: All dead codes removed.

File: dae-alright/src/components/FormInput/FormInput.component.jsx (hunk 6 above)
Comment (Joggz) [reply dae-alright/src/components/FormInput/FormInput.component.jsx]: Done.

File: dae-alright/src/components/FormInput/FormInput.component.jsx (hunk 7 above)
Comment (Joggz) [reply dae-alright/src/components/FormInput/FormInput.component.jsx]: added.

File: dae-alright/src/App.js (hunk 8 above)
Comment (Joggz) [reply dae-alright/src/App.js]: Removed.

**Review by Mlg29 — APPROVED** [review]
LGTM

**Review by SEUNAGBEYE — APPROVED** [review]
LGTM

**Pull request merged**